### PR TITLE
Adding zone distribution mode in the Cluster resource for Memorystore Redis cluster

### PR DESCRIPTION
**Pull request by vickramp**
Adding `zone_distribution_config` field to `google_redis_cluster` to allow creation of single zone clusters.

If this PR is for Terraform, I acknowledge that I have:

 - Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
 -  Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests). \
- [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [make test and make lint](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests. \
- [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

```release-note:enhancement
redis: added `zone_distribution_config` field to `google_redis_cluster`
```

### CODE REVIEW

**Review by SarahFrench — COMMENTED** [review]
Hi @vickramp, thanks for your PR!

Is the zoneDistributionConfig field present in the GA and/or Beta versions of the Memorystore API? Or is it in allowlist/private GA? I can't see the field in the public documentation and when the acceptance tests run the field is accepted by the API but isn't present when reading the resource's data after creation completes.

**Comment by vickramp** [comment]
Hi @SarahFrench This is for a new feature that is going to GA soon, we currently have the api fields under a visibility label which will be soon removed for launch.  I've now temporarily added the visibility label to the worker running these acceptance tests.

File: mmv1/products/redis/Cluster.yaml
Comment (SarahFrench) [inline mmv1/products/redis/Cluster.yaml:140]: nit: You could set `immutable: true` on the top-level field and it would [mark all subfields as immutable too](https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#immutable). It depends on whether this block is likely to contain a mixture of immutable and non-immutable fields in future or not.

File: mmv1/products/redis/Cluster.yaml
Comment (SarahFrench) [inline mmv1/products/redis/Cluster.yaml:141]: The documentation won't automatically provide information to users about a field being immutable. You could put `Immutable.` at the start of the field, and subfield, description(s) to make this clearer to users when they read the docs. [Here's an example](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudbuild_bitbucket_server_config#host_uri) of this being done for an immutable field in other resource.

**Review by SarahFrench — COMMENTED** [review]
I'm happy to approve this but I'm going to hold off until I get a 👍 from the Terraform provider team. I'm a HashiCorp employee and I'm still learning about how to handle PR approvals in the context of something that not yet in GA and going through Google's different launch stages.

I left a few comments about improving documentation of immutable fields that will help users, and a nit that isn't a blocker.

Note: I'm not sure why the missing test detector is saying that the new field isn't present in tests, as it's present in generated and handwritten tests. The handwritten test is Beta-specific, which may be a factor. Either way I'm satisfied with the test coverage and this isn't a blocker.

File: mmv1/products/redis/Cluster.yaml
Comment (vickramp) [reply mmv1/products/redis/Cluster.yaml:141]: ACK

File: mmv1/products/redis/Cluster.yaml
Comment (vickramp) [reply mmv1/products/redis/Cluster.yaml:140]: ACK

**Comment by vickramp** [comment]
> I'm happy to approve this but I'm going to hold off until I get a 👍 from the Terraform provider team. I'm a HashiCorp employee and I'm still learning about how to handle PR approvals in the context of something that not yet in GA and going through Google's different launch stages.
> 
> I left a few comments about improving documentation of immutable fields that will help users, and a nit that isn't a blocker.
> 
> Note: I'm not sure why the missing test detector is saying that the new field isn't present in tests, as it's present in generated and handwritten tests. The handwritten test is Beta-specific, which may be a factor. Either way I'm satisfied with the test coverage and this isn't a blocker.

Thanks Appreciate the help. 
Note: i will only merge this pull request once all the restrictions on the API are removed and is officially in GA. i have sent this for review so that i can launch API, gcloud and terraform everything at once, instead of waiting for approvals after the API is in GA.

**Comment by rileykarson** [comment]
Google-internal details captured in b/335469333, @vickramp will follow up here when we should merge.

**Review by SarahFrench — APPROVED** [review]
Approving, and will merge when given the heads up on here! Thanks all

**Review by rileykarson — CHANGES_REQUESTED** [review]
Just clicking this button so merge is blocked in the meantime. Any maintainer can manually dismiss.

**Comment by vickramp** [comment]
This PR is ready to be submitted, i have resolved any conflicts/ test failures.
FYI @rileykarson 

> Just clicking this button so merge is blocked in the meantime. Any maintainer can manually dismiss. 

@SarahFrench  As quoted by @rileykarson , he is fine with merging this change, so please merge this pr if you have no  other questions.

**Pull request merged**
